### ISSUE TITLE
Should cdi-api dependency be provided?

### ISSUE THREAD

**Issue by davidmelia** [derjust/spring-data-dynamodb#241]
Hi,

In version 5.0.4 we have the CDI 'compile' dependency :
```
        <dependency>
            <groupId>javax.enterprise</groupId>
            <artifactId>cdi-api</artifactId>
            <version>${cdi.version}</version>
            <scope>compile</scope>
        </dependency>
```

This dependency also pulls in 3 other dependencies included javax ex-api which can cause issues when using the hibernate validator.

I think this dependency should be <scope>provided</scope> as most people will not use CDI?

What do you think? 

Thanks

**Comment by derjust** [comment]
I agree that the dependencies can be a pain - especially as CDI is used by way fewer users.

Mid-term I'd like to separate things in three 'parts'
* Core / Spring
* CDI
* Spring-Boot
(My job doesn't allow for much work on the side right now but is improving at the end of the month significantly)
